src/Blocks/components/form/assets/utils.js

```diff
@@ -1247,6 +1251,91 @@ export class Utils {
 		this.conditionalTags.setField(formId, name);
 	}
 
+	/**
+	 * Set output results.
+	 *
+	 * @param {string} formId Form Id.
+	 * @param {object} response Api response.
+	 *
+	 * @returns {void}
+	 */
+	setResultsOutput(formId, data) {
+		// Check if we have output element - block.
+		const outputElement = document.querySelector(`${this.state.getStateSelector('resultOutput', true)}[${this.state.getStateAttribute('formId')}="${formId}"]`);
+
+		// If no output element, bailout.
+		if (!outputElement) {
+			return;
+		}
+
+		this.resetResultsOutput(formId);
+
+		// Check if we have output items.
+		const outputItems = data?.[this.state.getStateResponseOutputKey('resultOutputItems')] ?? {};
+
+		if (Object.keys(outputItems).length) {
+			for(const [key, value] of Object.entries(outputItems)) {
+				const itemElement = outputElement.querySelectorAll(`${this.state.getStateSelector('resultOutputItem', true)}[${this.state.getStateAttribute('resultOutputItemKey')}="${key}"][${this.state.getStateAttribute('resultOutputItemValue')}="${value}"]`);
+
+				if (itemElement.length) {
```

Comment:
Don't need the if here, `forEach` will skip empty arrays

Reply:
<img width="854" alt="image" src="https://github.com/infinum/eightshift-forms/assets/77000136/fee76575-9c86-40df-85dd-a5c0ee89c856">
`querySelectorAll` always returns a value 😉